src/library_pthread.js

```diff
@@ -281,12 +281,8 @@ var LibraryPThread = {
           // Worker wants to postMessage() to itself to implement setImmediate()
           // emulation.
           worker.postMessage(d);
-#if expectToReceiveOnModule('onAbort')
-        } else if (cmd === 'onAbort') {
-          if (Module['onAbort']) {
-            Module['onAbort'](d['arg']);
-          }
-#endif
+        } else if (cmd === 'callHandler') {
+          Module[d['handler']](...d['args']);
```

Comment:
What if the handler is not present?  (`expectToReceiveOnModule` just means that it could be present)

Reply:
The `Module[d['handler']]` check is not necessary here since it's checked below with `Module.hasOwnProperty(handler)`, and likewise for those `expectToReceiveOnModule` checks. Thus, this handler is called only if it's present in `INCOMING_MODULE_JS_API` and if it's explicitly set on the module.